File: src/Messaging/NBB.Messaging.Effects/MessageBus.cs
// Copyright (c) TotalSoft.
// This source code is licensed under the MIT license.

using NBB.Core.Effects;
using NBB.Messaging.Abstractions;

namespace NBB.Messaging.Effects
{
    public static class MessageBus
    {
        public static Effect<Unit> Publish(object message, MessagingPublisherOptions options = null) => Effect.Of<PublishMessage.SideEffect, Unit>(new PublishMessage.SideEffect(message, options));
    }
}

File: samples/MicroServices/NBB.Invoices.FSharp/NBB.Invoices.FSharp/Application/MessageBus.fs
// Copyright (c) TotalSoft.
// This source code is licensed under the MIT license.

namespace NBB.Invoices.FSharp.Application

open NBB.Core.Effects.FSharp

// TODO: Find a place for MesageBus wrapper
module MessageBus =
    let publish (obj: 'TMessage) =
        NBB.Messaging.Effects.MessageBus.Publish(obj :> obj)
        |> Effect.ignore

